### PR TITLE
Feat add TTL to environment resource

### PR DESCRIPTION
**Pull request by alonnoga**
### Issue & Steps to Reproduce / Feature Request
[//]: <> (choose “resolves” “fixes” or “closes” and put link for the issue)
Adding ttl to environment resource
1. read ttl
2. update ttl
3. when removing ttl it should be updated to infinite

### CODE REVIEW

**Comment by alonnoga** [comment]
[Environment Resource - TTL](https://trello.com/c/ZB4Q3QGZ/4837-environment-resource-ttl)

File: env0/resource_environment.go [hunk 1]
```diff
@@ -86,6 +88,20 @@ func resourceEnvironment() *schema.Resource {
...
+			"ttl": {
+				Type:        schema.TypeString,
+				Description: "the date the environment should be destroyed at (iso format)",
+				Optional:    true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					utcPattern := `\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d`
+					ttl := val.(string)
+					matched, err := regexp.MatchString(utcPattern, ttl)
+					if !matched || err != nil {
+						errs = append(errs, fmt.Errorf("%q must be of iso format, got: %q", key, ttl))
```
Comment (HeverFarber) [inline env0/resource_environment.go:100]: Maybe add example to ISO date to the error

**Review by HeverFarber — APPROVED** [review]
Nice!

File: env0/resource_environment.go (hunk 1 above)
Comment (alonnoga) [reply env0/resource_environment.go]: 4d01ba2 👍🏻

**Pull request merged**